docs/reference/src/main/asciidoc/injection.asciidoc

```diff
@@ -614,10 +614,40 @@ class OrderService {
                 handler.destroy();
             }
         }
+    }
   }
-}
 ----------------------------------------------------------------------------------
 
+Third method is a twist on the `select()` method, but it accepts `java.lang.reflect.Type` as parameter and optionally qualifier(s).
+This allows for generic selection of instances which can be handy while dealing with third party beans through extensions.
+However, in order to stay type-safe, this method has a limitation - it can only be invoked on `WeldInstance<Object>`.
+Invocation on any other type than `Object` will result in an  `IllegalStateException`.
+Please note that the return value if such select will always be `WeldInstance<Object>` unless you specify it further using `<SomeType>` before invoking this `select()`.
+Let's look at actual code:
```

Comment:
@mkouba if you have any better sample to present, I am all ears. I was unable to come up with other simple-enough case for this use case.